### DISCUSSION TITLE
Make route ID optional for @remix-run/testing createRemixStub

### DISCUSSION THREAD

**Discussion by kentcdodds** [remix-run/remix#5122] [Proposals]
Right now, if you don't pass a route.id, both the `routes` and `routeModules` properties get `undefined` keys that override one another for all the routes. TypeScript does't complain about this so we've got two options:

1. default the route ID to the path
2. require a route ID

I prefer option 1.

[Here's the code I'm talking about](https://github.com/remix-run/remix/blob/92bc8987f549db0f4ebd841076fe7a8fa02c3d0a/packages/remix-testing/create-remix-stub.tsx#L100):

<img width="707" alt="image" src="https://user-images.githubusercontent.com/1500684/213037040-2817fdc9-c209-469a-a6c7-9331096c305d.png">

**Comment by jrestall** [comment]
Would be great to see this fixed as useMatches is currently broken for me with @remix-run/testing if a route id isn't explicitly set as it tries to get a route handle [here](https://github.com/remix-run/remix/blob/92bc8987f549db0f4ebd841076fe7a8fa02c3d0a/packages/remix-react/components.tsx#L906). I hit this very often since remix-validated-form uses useMatches internally.

In remix-stubs I call [convertRoutesToDataRoutes](https://github.com/remix-run/react-router/blob/4f3ad7b96e6e0228cc952cd7eafe2c265c7393c7/packages/router/utils.ts#L274) via createStaticHandler which automatically sets unique ids for each route and avoids this issue.

So maybe there's an Option 3 where convertRoutesToDataRoutes is used to default all the route ids to their tree path.
